File: src/main/java/io/github/kuroka3/playermanager/Utils/BanIDManager.java
package io.github.kuroka3.playermanager.Utils;

import io.github.kuroka3.playermanager.PlayerManager;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

import java.io.IOException;
import java.time.LocalDateTime;
import java.util.UUID;

public class BanIDManager {

    private static final JSONParser parser = new JSONParser();

    public static void setBan(String id, UUID target, String moder, String reason, LocalDateTime time) {

        JSONFile file = new JSONFile(PlayerManager.getPlugin(PlayerManager.class).getDataFolder() + "/banid.json");

        JSONObject fullJson = loadJSON(file);

        JSONObject banInfo = new JSONObject();

        banInfo.put("target", target.toString());
        banInfo.put("moder", moder);
        banInfo.put("reason", reason);
        banInfo.put("time", time.toString());

        fullJson.put(id, banInfo);

        saveJSON(fullJson, file);
    }

    public static @Nullable JSONObject getBanInfo(String id) {

        JSONFile file = new JSONFile(PlayerManager.getPlugin(PlayerManager.class).getDataFolder() + "/banid.json");

        JSONObject jsonObject = loadJSON(file);

        return (JSONObject) jsonObject.get(id);

    }

    private static @NotNull JSONObject loadJSON(JSONFile jsonFile) {
        try {
            if (!jsonFile.isFile()) {
                if(!jsonFile.createNewFile()) {
                    PlayerManager.getPlugin(PlayerManager.class).getDataFolder().mkdir();
                    jsonFile.createNewFile();
                }
                return new JSONObject();
            }

            return jsonFile.getJSONObject();

        } catch (Exception e) {
            e.printStackTrace();
            return new JSONObject();
        }
    }

    private static void saveJSON(JSONObject jsonObject, JSONFile jsonFile) {
        try {
            jsonFile.saveJSONFile(jsonObject);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}

File: src/main/kotlin/io/github/kuroka3/playermanagerkotlin/Utils/BanIDManager.kt
package io.github.kuroka3.playermanagerkotlin.Utils

import io.github.kuroka3.playermanagerkotlin.PlayerManagerKotlin
import org.json.simple.JSONObject
import java.io.IOException
import java.time.LocalDateTime
import java.util.*

object BanIDManager {
    lateinit var banidJSONFile: JSONFile

    fun setInit() {
        banidJSONFile = JSONFile("${PlayerManagerKotlin.instance.dataFolder}/data/banid.json")
    }
    
    fun setBan(id: String?, target: UUID, moder: String?, reason: String?, time: LocalDateTime, istemp: Boolean?) {
        setBan(id, target, moder, reason, time, istemp, null)
    }

    fun setBan(
        id: String?,
        target: UUID,
        moder: String?,
        reason: String?,
        time: LocalDateTime,
        istemp: Boolean?,
        period: String?
    ) {
        val file = banidJSONFile
        val fullJson = loadJSON(file)
        val banInfo = JSONObject()
        banInfo["target"] = target.toString()
        banInfo["moder"] = moder
        banInfo["reason"] = reason
        banInfo["time"] = time.toString()
        banInfo["isTemp"] = istemp
        if (period != null) banInfo["period"] = period
        fullJson[id] = banInfo
        saveJSON(fullJson, file)
    }

    fun getBanInfo(id: String): JSONObject? {
        val file = banidJSONFile
        val jsonObject = loadJSON(file)
        return jsonObject[id] as JSONObject?
    }

    private fun loadJSON(jsonFile: JSONFile): JSONObject {
        return try {
            if (!jsonFile.isFile) {
                if (!jsonFile.createNewFile()) {
                    PlayerManagerKotlin.instance.dataFolder.mkdir()
                    jsonFile.createNewFile()
                }
                return JSONObject()
            }
            jsonFile.jsonObject!!
        } catch (e: Exception) {
            e.printStackTrace()
            JSONObject()
        }
    }

    private fun saveJSON(jsonObject: JSONObject, jsonFile: JSONFile) {
        try {
            jsonFile.saveJSONFile(jsonObject)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}
